driver/bpf/fillers.h

```diff
@@ -2151,20 +2151,32 @@ static __always_inline bool bpf_kgid_has_mapping(struct user_namespace *targ, kg
 	return bpf_map_id_up(&targ->gid_map, __kgid_val(kgid)) != (gid_t) -1;
 }
 
-static __always_inline bool get_exe_writable(struct task_struct *task)
+static __always_inline struct inode *get_exe_inode(struct task_struct *task)
 {
 	struct file *exe_file;
 	struct mm_struct *mm;
+	
 	mm = _READ(task->mm);
```

Comment:
Agree with @incertum here we need to choose a way to handle this `NULL` check. To be honest, I would avoid the check unless it is necessary or explicitly required by the verifier... every check has a ridiculous cost but if we put many of them in the same flow it could impact perf.

Reply:
and adding them we increase also the bytecode complexity with many jumps

Reply:
Until we use the pointer only to perform another `_READ` we should be safe without the `NULL` check, this is the same logic I've followed in the modern probe, obviously here we are talking about lower kernel versions so never say never :roll_eyes: